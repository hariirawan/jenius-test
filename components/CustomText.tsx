import { View, Text, StyleSheet, TextProps } from "react-native";
import React from "react";
import { Colors } from "@/constants/Colors";

interface IText {
  color?: "primary" | "secondary" | "light" | "tertiary";
  type: "title" | "subtitle" | "caption" | "text";
  children?: React.ReactNode;
  style?: TextProps;
}

export default function CustomText(props: IText) {
  return (
    <Text
      style={[
        props.type === "title" ? styles.title : null,
        props.type === "subtitle" ? styles.subtitle : null,
        props.type === "caption" ? styles.caption : null,
        props.type === "text" ? styles.text : null,
        { color: Colors[props.color ?? "primary"] },
      ]}
    >
      {props.children}
    </Text>
  );
}

const styles = StyleSheet.create({
  title: {
    fontFamily: "Poppins-SemiBold",
    fontSize: 20,
    color: "#000",
  },
  subtitle: {
    fontFamily: "Poppins-Regular",
    fontSize: 16,
    color: "#000",
  },
  caption: {
    fontFamily: "Poppins-Medium",
    fontSize: 14,
    color: "#000",
  },
  text: {
    fontFamily: "Poppins-Light",
    fontSize: 12,
    color: "#000",
  },
});
