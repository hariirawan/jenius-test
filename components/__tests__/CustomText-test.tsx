import * as React from "react";
import renderer from "react-test-renderer";

import CustomText from "../CustomText";
import { Text } from "react-native";

describe("CustomText", () => {
  it("renders with default props", () => {
    const tree = renderer
      .create(<CustomText>Default text</CustomText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom color", () => {
    const tree = renderer
      .create(<CustomText color="secondary">Custom color text</CustomText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom type", () => {
    const tree = renderer
      .create(<CustomText type="title">Title text</CustomText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with custom style", () => {
    const customStyle = { fontSize: 24, fontWeight: "bold" };
    const tree = renderer
      .create(<CustomText style={customStyle}>Custom style text</CustomText>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders with children", () => {
    const tree = renderer
      .create(
        <CustomText>
          <Text>Hello</Text> <Text>World!</Text>
        </CustomText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("overrides default color with custom color", () => {
    const tree = renderer
      .create(
        <CustomText color="tertiary" type="title">
          Custom color title text
        </CustomText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
