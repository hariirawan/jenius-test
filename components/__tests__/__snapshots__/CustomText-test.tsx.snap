// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText overrides default color with custom color 1`] = `
<Text
  style={
    [
      undefined,
      {
        "color": "#000",
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 20,
      },
      null,
      null,
      null,
      {
        "color": "#F8D51C",
      },
    ]
  }
>
  Custom color title text
</Text>
`;

exports[`CustomText renders with children 1`] = `
<Text
  style={
    [
      undefined,
      null,
      null,
      null,
      null,
      {
        "color": "#0E243C",
      },
    ]
  }
>
  <Text>
    Hello
  </Text>
   
  <Text>
    World!
  </Text>
</Text>
`;

exports[`CustomText renders with custom color 1`] = `
<Text
  style={
    [
      undefined,
      null,
      null,
      null,
      null,
      {
        "color": "#8BD2BD",
      },
    ]
  }
>
  Custom color text
</Text>
`;

exports[`CustomText renders with custom style 1`] = `
<Text
  style={
    [
      {
        "fontSize": 24,
        "fontWeight": "bold",
      },
      null,
      null,
      null,
      null,
      {
        "color": "#0E243C",
      },
    ]
  }
>
  Custom style text
</Text>
`;

exports[`CustomText renders with custom type 1`] = `
<Text
  style={
    [
      undefined,
      {
        "color": "#000",
        "fontFamily": "Poppins-SemiBold",
        "fontSize": 20,
      },
      null,
      null,
      null,
      {
        "color": "#0E243C",
      },
    ]
  }
>
  Title text
</Text>
`;

exports[`CustomText renders with default props 1`] = `
<Text
  style={
    [
      undefined,
      null,
      null,
      null,
      null,
      {
        "color": "#0E243C",
      },
    ]
  }
>
  Default text
</Text>
`;
